File: DSA/DevvratKhandal_590014906/Day_10/Question_1.java
import java.util.Scanner;

public class Question_1 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter haystack: ");
        String haystack = sc.nextLine();

        System.out.print("Enter needle: ");
        String needle = sc.nextLine();

        int ans = -1;

        for (int i = 0; i <= haystack.length() - needle.length(); i++) {
            boolean found = true;

            for (int j = 0; j < needle.length(); j++) {
                if (haystack.charAt(i + j) != needle.charAt(j)) {
                    found = false;
                    break;
            }
            }

            if (found) {
                ans = i;
                break;
        }
        }

        System.out.println("Index = " + ans);

        sc.close();
}
}

File: DSA/NabhanshRishiGaur_590028370/Day_10/Question_1.c
int strStr(char* haystack, char* needle) {

    int l1 = strlen(haystack);
    int l2 = strlen(needle);

    int i = 0;
    int j = 0;
    int idx = -1;

    for (i = 0; i <= l1 - l2; i++) {
        j = 0;
        while ( j< l2 && (haystack[i+j] == needle[j])) {
            j++;
        }
        if(j == l2) 
            return i;
        
    }
return -1;
}
